packages/dd-trace/src/proxy.js

```diff
@@ -117,10 +117,11 @@ class Tracer extends NoopProxy {
       if (config.iast.enabled) {
         require('./appsec/iast').enable(config, this._tracer)
       }
-    } else {
+    } else if (this._tracingInitialized) {
```

Comment:
why are you checking this var ?

Reply:
if appsec and iast were ever required, `this._tracingInitialized` is set to true

Reply:
no it's not ?

Reply:
> no it's not ?

Can you clarify exactly what you are saying is not working? What I see is that:

1) If `tracing` is set to `false` and it was previously set to `false`, then the code to set `_tracingInitialized` would never have run and the variable wouldn't be set, so this condition wouldn't run and AppSec would never be imported, as expected.
2) If `tracing` is set to `false` and it was previsouly set to `true`, then the code to set `_tracingInitialized` will have run, so this condition will run and AppSec will already have been imported before anyway by the same code that set `_tracingInitialized`, and will now be disabled, as expected.

It seems to be that this is working as intended. Am I missing something?

Reply:
Your 1. sounds correct to me, but for 2. I don't follow your logic.

What I see is:
-  `_tracingInitialized` is set to true in line 115 and this line only.
- AppSec is required and enabled in line 110 and this line only.
They are not in the same branch.

Reply:
So from what I read, if appsec was never required nor enabled, but tracing is toggled from on to off, it will magically require appsec to try to disable it, even tho it was never required nor enabled, and exposes the app to potential instabilities caused by appsec that the user never even asked for.

Reply:
> _tracingInitialized is set to true in line 115 and this line only.
AppSec is required and enabled in line 110 and this line only.

Yes, and both only run together and when tracing is enabled. So unless that ever ran, disabling AppSec will not run.

Reply:
Ok I see it now, it's if `appsec.enabled` is false when `tracing` is true, good catch.